pkgs/os-specific/linux/kinect-audio-setup/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, stdenv
+, fetchgit
+, requireFile
+, pkg-config
+, libusb1
+, p7zip
+}:
+
+let
+  firmwareUrl = "https://download.microsoft.com/download/F/9/9/F99791F2-D5BE-478A-B77A-830AD14950C3/KinectSDK-v1.0-beta2-x86.msi";
+  licenseUrl = "https://research.microsoft.com/en-us/um/legal/kinectsdk-tou_noncommercial.htm";
+in
+stdenv.mkDerivation rec {
+  pname = "kinect-audio-setup";
+  version = "0.5";
+
+  FIRMWARE = requireFile rec {
+    name = "UACFirmware";
+    sha256 = "08a2vpgd061cmc6h3h8i6qj3sjvjr1fwcnwccwywqypz3icn8xw1";
+    message = ''
+      In order to install the Kinect Audio Firmware, you need to download the
+      non-redistributable firmware from Microsoft.
+      The firmware is available at ${firmwareUrl} and the license at ${licenseUrl} .
+      Save the file as UACFirmware and use "nix-prefetch-url file://\$PWD/UACFirmware" to
+      add it to the Nix store.
+    '';
+  };
+
+  src = fetchgit {
+    url = "git://git.ao2.it/kinect-audio-setup.git";
+    rev = "v${version}";
+    sha256 = "sha256-bFwmWh822KvFwP/0Gu097nF5K2uCwCLMB1RtP7k+Zt0=";
+  };
+
+  patches = [
+    ./libusb-1-import-path.patch
+    ./udev-rules-extra-devices.patch
```

Comment:
It would be good to have a short comment for each patch that explains its purpose. You can already see that when you open the patch, but I think it is nice to have each description available on a glance. (This is a "nice to have", not a requirement for merge).

Reply:
Is there a reason that the patches cannot be upstreamed? If not, have you proposed them upstream? Please add a reference if you have done that.

Reply:
Hi,

The patches already contain comments (`git diff -- file > patch`).

The project is inactive since 2016 so I did not attempt to upstream the patches.

Reply:
I'm aware that the patches contain the comments, I just think it is more convenient to have that information right next to the patch name. But as I said, that's not mandatory.

It is a shame that the project is inactive. It makes me a bit hesitant about merging. It has very little logic and is still relevant though, so it is probably okay in this case.